### DISCUSSION TITLE
Maximum time for opening channel elapsed, Channel stuck in waiting close

### DISCUSSION THREAD

**Discussion by arman-ta** [lightningnetwork/lnd#8931] [Q&A]
Last year I 've initiated opening a channel and unfortunately it was with low fee and I forgot about it. The maximum 2016 block have elapse and the transaction got the confirmation after more than 2 months!

I have been backing up the SCB and my node crashed. I tried to restore the node using my seed and SCB. All my fund was restored and my open channels got force closed, except this channel which I have 1 mil sats stuck in there. The channel is not shown on the network as either open nor closed, but the status in my node is stuck as waiting close. Capacity is 1 mil sats but local and remote balance are shown 0. 
![Screenshot 2024-07-16 233845](https://github.com/user-attachments/assets/6b9c55b1-44a0-4294-9dca-3685a45e7122)
![Screenshot 2024-07-17 233413](https://github.com/user-attachments/assets/ab486407-3776-4f53-b393-8fc4fde88df1)

The Peer node has been offline for few month now and there is no way I can make contact, even email messages do not deliver to the peer!

Is there any way I could recover my stuck fund in this case?

I really appreciate if anyone could support me, and definitely willing to give away a portion of it as a gratitude if I can recover my fund!

**Comment by guggero** [comment]
Can you please type the "Channel Point" as text here so we don't have to type it from the screenshot to look it up on chain?

**Comment by arman-ta** [comment]
@guggero Thanks a lot for getting back to me!

Channel point: `ed76e2777c449526cfc581c63a9226a71e0854c362984b0391cb4a2012d69b7e:0`
Peer Pubkey: `02909a6c285e21f952acc9cc51ec1c4abdf6295452436fde5e5b79a16d4a5e7ef9`
Peer ip address: `z3htoq3aavowdz75n74gofxavi7duiywbypewaw2vpjeo766iiby25ad.onion:9735`

My  Pubkey: `02fc1f8c244c9d275f999e6f0ce02f8e46d8f31864e9c899392dac5260ee45d9b7`

I used `chantools triggerforceclose` but I get the following error:

![Screenshot 2024-07-19 150443](https://github.com/user-attachments/assets/439bffef-dfee-4d9b-b20a-1e70ce3ede04)

**Reply by guggero** [reply]
Thanks for the info. So yeah, unfortunately the transaction did make it into the chain. But because it took too long, the peer did remove the data from their database. So even if the `chantools triggerforceclose` command worked (you need to run Tor locally), it wouldn't do anything.
By restoring your node from SCB you deleted the only state left that could've closed the channel. Using SCB means relying on your peers to close. But because for this channel the peer didn't keep the state, nobody can close it now.

Unfortunately there are only two ways to recover these funds:
 - Find a way to contact the person operating the node on the other end ([`Bitcoinakademin`](https://1ml.com/node/02909a6c285e21f952acc9cc51ec1c4abdf6295452436fde5e5b79a16d4a5e7ef9), then go through the `chantools zombierecovery` process with them.
 - If you did make a file based backup of your node before restoring from seed, you can run `chantools forceclose` on the channel. But that's publishing an old state, so it's risky if the other node is still running (which it seems they might not be).

**Reply by TrezorHannes** [reply]
> * Find a way to contact the person operating the node on the other end ([`Bitcoinakademin`](https://1ml.com/node/02909a6c285e21f952acc9cc51ec1c4abdf6295452436fde5e5b79a16d4a5e7ef9), then go through the `chantools zombierecovery` process with them.
> * If you did make a file based backup of your node before restoring from seed, you can run `chantools forceclose` on the channel. But that's publishing an old state, so it's risky if the other node is still running (which it seems they might not be).

Hi @guggero 

I'm trying to help Arman with the recovery. Thanks for looking into this for us.
Just to validate not missing anything:
- we need to assume the peer is offline. But even if we would be able to reach them, since the channel never confirmed within 2016 blocks, they wouldn't be able to run the chantools zombierecovery collab anyway, wouldn't they? Since their node forgot everything
- we don't have an outdated / pre-SCB channel.db, just a bunch of SCBs from that time. Is there anything we could do with this?
- since we can assume the peer is gone for long, and can safely assume Arman has no malicious intend since he was the opener and the confirmation took 3 months: Isn't there a way to alter the local state? Under usual circumstances, that would result in a penalty tx, but since he got 0 state, is that even feasible?

The json output of `chantools summary` looks like this, so 0 liq on both ends
![image](https://github.com/user-attachments/assets/4b60820a-9391-4269-81dc-cf2a69ee99ae)

Thanks again

**Comment by ziggie1984** [comment]
> we need to assume the peer is offline. But even if we would be able to reach them, since the channel never confirmed within 2016 blocks, they wouldn't be able to run the chantools zombierecovery collab anyway, wouldn't they? Since their node forgot everything

Correct, because the channel was never published the tool might never find the match, however you can fake the match file and just inject the pubkey of the peers node, through this it's still possible to find the channel (assuming both nodes didn't open more than 2500 channels, which is the default number of keys which will be derived), the fake match file should look like this:

```
{
  "node1": {
    "identity_pubkey": "PUBKEY_OF_NODE1",
    "contact": "",
    "payout_addr": "",
    "multisig_keys": [
    ]
  },
  "node2": {
    "identity_pubkey": "PUBKEY_OF_NODE2",
    "contact": "",
    "multisig_keys": [
    ]
  },
  "channels": [
  ]
}

```

=> save it and provide it with the --match_file handle of the `preparekeys` command.

you need to provide this file also to the peer, which needs to run also  the `preparekeys` cmd. (reach out to me if you have problems doing so.)

of-course one of the keys must match the node's pubkey.


> we don't have an outdated / pre-SCB channel.db, just a bunch of SCBs from that time. Is there anything we could do with this?

nope, the SCB has not channel-state therefore you cannot do anything even with prior SCBs.



> since we can assume the peer is gone for long, and can safely assume Arman has no malicious intend since he was the opener and the confirmation took 3 months: Isn't there a way to alter the local state? Under usual circumstances, that would result in a penalty tx, but since he got 0 state, is that even feasible?


No, the SCB does not mean it recreated any kind of channel state, rather its just a shell and you cannot create any prior or new states in this state.
